src/operator/nn/deconvolution-inl.h

```diff
@@ -143,9 +143,9 @@ struct DeconvolutionParam : public dmlc::Parameter<DeconvolutionParam> {
         }
       }
     } else {
-      for (size_t i = 0; i < ndim; i++) {
-        o_pad[i] = pad[i];
-        o_adj[i] = adj[i];
+      for (int i = 0; i < (int) ndim; i++) {
+        o_pad[i] = i < pad.ndim() ? pad[i] : 0;
```

Comment:
According the [initialization function](https://github.com/apache/incubator-mxnet/blob/master/src/operator/nn/deconvolution.cc#L335), `pad.ndim()` should be equal to `ndim` here. Do you understand why `i` can be bigger than `pad.ndim()`?

Reply:
The upsampling operator customized DeconvolutionParam [here](https://github.com/apache/incubator-mxnet/blob/master/src/operator/nn/upsampling-inl.h#L170-L188). It is kinda weird but I believe adj is not used in this case.